### PR TITLE
Jelmer Sander create states and screens and game and no Jody Bernal :(

### PR DESCRIPTION
**Pull request by waseland**
@waseland	waseland	feat: added DrawRect to SDL Facade and Raycasting Engine			9f6588a
 @SandervGeel	SandervGeel	feat: IDrawable			fdeb769
 @SandervGeel	SandervGeel	fix: IDrawable make error			8a91397
 @waseland	waseland	feat: Virtual and Deconstructor			1f9d0be
 @waseland	waseland	feat: draw_image in SDL_Facade			3770355
 @waseland	waseland	fix: removed draw_rect from Raycasting.hpp			4dd7fde
 @SandervGeel	SandervGeel	feat: state pattern set up			0926940
 @SandervGeel	SandervGeel	Merge branch 'JelmerSander_credits' of https://github.com/ThomasvLing…  …			7e4ac8c
 @SandervGeel	SandervGeel	fix: bug fix			0b8d6bd
 @waseland	waseland	fix: Squiggle lines and CMakeList error			82e65a4
 @SandervGeel	SandervGeel	feat: initialization states			7191a47
 @SandervGeel	SandervGeel	Merge branch 'JelmerSander_credits' of https://github.com/ThomasvLing…  …			e77509e
 @SandervGeel	SandervGeel	fix: error fix for pull request			ace87f9
 @waseland	waseland	feat: added credit and start screen and fixed memory leak in draw_tex…  …			f5e257a
Commits on Nov 18, 2016
 @SandervGeel	SandervGeel	feat: state pattern			51163fb
 @SandervGeel	SandervGeel	Merge branch 'JelmerSander_credits' of https://github.com/ThomasvLing…  …			1a91060
 @waseland	waseland	feat: added credit_screen			7d5cf69
 @waseland	waseland	Merge branch 'dev' of https://github.com/ThomasvLingen/Minor_vidya in…  …			d692f21
 @waseland	waseland	feat: made states working and made them work with each other			bfad9e4
 @SandervGeel	SandervGeel	feat: help state			8e7c1cd
 @SandervGeel	SandervGeel	feat: error fix includes			fed599a
 @waseland	waseland	feat: created help screen			cb0864a
 @waseland	waseland	fix: loading screen time and warning in bool Game::_running

### CODE REVIEW

File: src/gamelogic/State/CreditState.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    CreditState::~CreditState() {
+
+    }
+
+    void CreditState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
+        PressedKeys keys = game._SDL_facade.get_keys();
+        if(timeSinceLastPress <= 20){ //TODO: TO SDL ticks
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: I see this tidbit of code in 5 different places.

Something is wrong architecturally wise.

File: src/gamelogic/State/CreditState.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    class CreditState : public IGameState {
+
+    public:
+        CreditState();
+        ~CreditState();
+
+        void update(GameLogic::Game& game) override ;
+    private:
+        int timeSinceLastPress = 0;
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.hpp]: snake_case.

This should also be preceded with an underscore

File: src/gamelogic/State/CreditState.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    }
+
+    CreditState::~CreditState() {
+
+    }
+
+    void CreditState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: do we need the raycaster to do anything in the credits state?

File: src/gamelogic/State/CreditState.hpp (hunk 2 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.hpp]: Also, name is misleading. This is not the time since the last press, its the frames since the last press

File: src/gamelogic/State/CreditState.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        PressedKeys keys = game._SDL_facade.get_keys();
+        if(timeSinceLastPress <= 20){ //TODO: TO SDL ticks
+            timeSinceLastPress++;
+        }
+        if(timeSinceLastPress > 20){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: wut? Why are we doing this?

You're essentially delaying the processing of pressed keys by 1/3 second.

File: src/gamelogic/State/CreditState.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                        timeSinceLastPress  = 0;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        game._SDL_facade.clear_screen();
+        Color color{255,255,255};
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: It would make sense to define this outside of this function since it never changes and is a constant.

That way we don't have to make a Color object every time we get here.

File: src/gamelogic/State/CreditState.cpp [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        game._SDL_facade.draw_image("res/creditscreen.bmp" , CoordinateDouble{0,0});
+        game._SDL_facade.draw_text("Vidya Game is made by:", FontType::alterebro_pixel, color, CoordinateDouble{300, 50});
+        game._SDL_facade.draw_text("- Thomas Pannekoek:", FontType::alterebro_pixel, color, CoordinateDouble{340, 80});
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: `>.>`

File: src/gamelogic/State/HelpState.cpp [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+
+        game._SDL_facade.clear_screen();
+        Color color{255,255,255};
+        game._SDL_facade.draw_image("res/helpscreen.bmp" , CoordinateDouble{0,0});
+        game._SDL_facade.draw_text("Controls for Vidya Game", FontType::alterebro_pixel, color, CoordinateDouble{120, 50});
+        game._SDL_facade.draw_text("Movement Forward:   W", FontType::alterebro_pixel, color, CoordinateDouble{160, 80});
+        game._SDL_facade.draw_text("Movement Backward:  S", FontType::alterebro_pixel, color, CoordinateDouble{160, 110});
+        game._SDL_facade.draw_text("Movement Left:         A", FontType::alterebro_pixel, color, CoordinateDouble{160, 140});
```
Comment (ThomasvLingen) [inline src/gamelogic/State/HelpState.cpp]: All these spaces here look kind of odd. I'd say split the draw_texting of the text and the key to make them all line up instead.

File: src/gamelogic/State/IGameState.cpp [hunk 8]
```diff
@@ -0,0 +1,23 @@
+//
+// Created by sander on 11/17/16.
+//
+
+#include "IGameState.hpp"
+#include "../Game.hpp"
+
+namespace State {
+
+    IGameState::IGameState() {
```
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.cpp]: Empty const/destructors

File: src/gamelogic/State/IGameState.hpp [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace State{
+
+    class IGameState {
+
+    public:
+        IGameState();
+        virtual ~IGameState();
+
+        virtual void update(GameLogic::Game& game);
```
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.hpp]: Why not make this a pure virtual instead? You don't ever want to instantiate an IGameState or call it's update right?

File: src/gamelogic/State/IGameState.cpp [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+    }
+
+    IGameState::~IGameState() {
+
+    }
+
+    void IGameState::update(GameLogic::Game& game) {
```
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.cpp]: See comment in .hpp

File: src/gamelogic/State/LoadState.hpp [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+    class LoadState : public IGameState {
+
+    public:
+        LoadState();
+        ~LoadState();
+
+        void update(GameLogic::Game& game) override ;
+    private:
+        int timer = 0;
```
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.hpp]: precede with underscore

Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.hpp]: Also timer is a far too generic name. Try to make it clear with the name what purpose this variable serves

File: src/gamelogic/State/LoadState.cpp [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    LoadState::~LoadState(){
+
+    }
+
+    void LoadState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
+        game._SDL_facade.clear_screen();
+        game._SDL_facade.draw_rect(CoordinateDouble{0,0}, 640, 480, Color{0,0,0});
+        timer = timer + 2; //TODO: To ticks
```
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.cpp]: Why +2?

File: src/gamelogic/State/LoadState.cpp [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+        Color color{255,255,255};
+        game._SDL_facade.draw_text("Level Name Here", FontType::alterebro_pixel, color, CoordinateDouble{500,15});
+        if(timer < 150){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.cpp]: This if else if ladder is really ugly.

File: src/gamelogic/State/MenuState.cpp [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+    }
+
+    MenuState::~MenuState(){
+
+    }
+
+    void MenuState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: does that raycaster need to do anything?

File: src/gamelogic/State/MenuState.cpp [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+        PressedKeys keys = game._SDL_facade.get_keys();
+        if(timeSinceLastPress <= 20){ //TODO: TO SDL ticks
+            timeSinceLastPress++;
+        }
+        if(timeSinceLastPress > 20){
+            for(auto key : keys){
+                switch (key) {
+                    case Key::W :
+                        if(selected != 1){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: use this->

Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: magic number

File: src/gamelogic/State/MenuState.cpp [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+                            selected--;
+                            timeSinceLastPress = 0;
+                        }
+                        break;
+                    case Key::S :
+                        if(selected != 3){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: magic number

File: src/gamelogic/State/MenuState.cpp [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+                            selected++;
+                            timeSinceLastPress = 0;
+                        }
+                        break;
+                    case Key::ESC:
+                        game.set_new_state(game.get_start_up_state());
+                        timeSinceLastPress = 0;
+                        break;
+                    case Key::ENTER:
+                        if(selected == 1){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: another if else if ladder.
Not too sure what to do with these.

File: src/gamelogic/State/MenuState.cpp [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+        game._SDL_facade.clear_screen();
+        game._SDL_facade.draw_image("res/menuscreen.bmp", CoordinateDouble{0,0});
+        Color color{255,255,255};
+        game._SDL_facade.draw_text("Vidya Game", FontType::alterebro_pixel, color, CoordinateDouble{20,20});
+        game._SDL_facade.draw_text("Start Game", FontType::alterebro_pixel, color, CoordinateDouble{150,100});
+        game._SDL_facade.draw_text("Help", FontType::alterebro_pixel, color, CoordinateDouble{150,140});
+        game._SDL_facade.draw_text("Credits", FontType::alterebro_pixel, color, CoordinateDouble{150,180});
+
+
+        if(selected == 1){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: There must be a cleaner way to do this instead of iffing the selected.

Can't you just set the y position to `selected * magic_constant`?

File: src/gamelogic/State/MenuState.hpp [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+    class MenuState : public IGameState {
+
+    public:
+        MenuState();
+        ~MenuState();
+
+        void update(GameLogic::Game& game) override ;
+    private:
+        int timeSinceLastPress = 0;
+        int selected = 1;
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.hpp]: Why do we start with an index of 1 instead of an index of 0?

File: src/gamelogic/State/MenuState.cpp [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+
+    MenuState::MenuState(){
+
+    }
+
+    MenuState::~MenuState(){
+
+    }
+
+    void MenuState::update(GameLogic::Game& game) {
```
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: This state generally has a lot of Menu logic. Wouldn't it be better for seperation of concerns if the Menustate controls some Menu object and delegates things to it? (like select next, select previous, click_item(selected)) etc?

File: src/gamelogic/State/PauseState.cpp [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+
+    }
+
+    PauseState::~PauseState() {
+
+    }
+
+    void PauseState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
```
Comment (ThomasvLingen) [inline src/gamelogic/State/PauseState.cpp]: the raycaster shouldn't do jack whilst paused, right?

File: src/gamelogic/State/PauseState.cpp [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+        if(timeSinceLastPress <= 15){ //TODO: TO SDL ticks
```
Comment (ThomasvLingen) [inline src/gamelogic/State/PauseState.cpp]: why is this number different from the others?
Also, code duplication

File: src/gamelogic/State/PauseState.cpp [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+            timeSinceLastPress++;
+        }
+        if(timeSinceLastPress >= 15) {
+            PressedKeys keys = game._SDL_facade.get_keys();
+            for (auto key : keys) {
+                switch (key) {
+                    case Key::W:
+                        if(selected != 1){
```
Comment (ThomasvLingen) [inline src/gamelogic/State/PauseState.cpp]: Hey, duplicate menu code, this is generally why you want to encapsulate actual menu code into a class instead.

File: src/gamelogic/State/StartUpState.cpp [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+    }
+
+    StartUpState::~StartUpState(){
+
+    }
+
+    void StartUpState::update(GameLogic::Game& game) {
+        game._SDL_facade.handle_sdl_events();
+        game._raycasting_engine.handle_input();
```
Comment (ThomasvLingen) [inline src/gamelogic/State/StartUpState.cpp]: Again, does the raycaster have to do osmething here?

File: src/engine/SDLFacade.cpp [hunk 25]
```diff
@@ -131,6 +131,56 @@ namespace Engine {
...
+    ///
+    /// A rect will be drawn from the given coordinate with the given width and height The color is set first, secondly the rect is being drawn.
+    ///
+    /// \param rect_start Coordinate of the start of the rectangle
+    /// \param width Width of the rectangle
+    /// \param height Height of the rectangle
+    /// \param color The color of the rectangle. The color class comes from the engine
+    void SDLFacade::draw_rect(const CoordinateDouble& rect_start, const int width, const int height, const Color& color)
+    {
+        SDL_SetRenderDrawColor(this->_renderer, color.r_value, color.g_value, color.b_value, 255);
```
Comment (waseland) [inline src/engine/SDLFacade.cpp]: Oops...

File: src/engine/SDLFacade.cpp [hunk 26]
```diff
@@ -99,8 +99,8 @@ namespace Engine {
     /// \return This function returns True if the _window was successfully initialized, ohterwise it returns False
     bool SDLFacade::_init_window()
     {
-        this->_window = SDL_CreateWindow("Vidya game", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 500,
-                                   500, SDL_WINDOW_SHOWN);
+        this->_window = SDL_CreateWindow("Vidya game", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, this->_width,
```
Comment (waseland) [inline src/engine/SDLFacade.cpp]: Uhh... Well you saw what happened when I put this back to 500x500. No cropped images at this time!

File: src/engine/SDLFacade.cpp [hunk 27]
```diff
@@ -131,6 +131,56 @@ namespace Engine {
...
+    /// \brief Draws a image
+    ///
+    /// An image will be drawn on the coordinates (Accepts only bmp files)
+    ///
+    /// \param path Path to image (if in res it is: res/imagename.bmp)
+    /// \param coordinates Coordinates of where the image has to be drawn
+    void SDLFacade::draw_image(const std::string path, const CoordinateDouble& coordinates)
+    {
+        const char* imagePath = path.c_str();
+        SDL_Surface* image = SDL_LoadBMP(imagePath);
```
Comment (waseland) [inline src/engine/SDLFacade.cpp]: Seems fine to me

File: src/engine/SDLFacade.cpp [hunk 28]
```diff
@@ -243,6 +294,7 @@ namespace Engine {
     bool SDLFacade::_init_fonts()
     {
         //todo Needs to be expanded later on
+        _load_font("res/alterebro_pixel.ttf", FontType::alterebro_pixel_plus, 60);
```
Comment (waseland) [inline src/engine/SDLFacade.cpp]: Solved with bool &=

File: src/engine/raycasting/Raycasting.hpp [hunk 29]
```diff
@@ -19,7 +19,6 @@ namespace Engine {
     class Raycasting {
     public:
         Raycasting(SDLFacade& SDL_facade);
-
```
Comment (waseland) [inline src/engine/raycasting/Raycasting.hpp]: Is this in the programming guidelines?

File: src/gamelogic/Game.cpp [hunk 30]
```diff
@@ -67,29 +83,25 @@ namespace GameLogic {
...
+    Game::~Game() {
+
+    }
+
     /// \brief The main game loop
     ///
     /// first handle input, thereafter update and finally clear and draw the screen.
     void Game::run()
     {
-        int time_since_last_frame;
```
Comment (waseland) [inline src/gamelogic/Game.cpp]: Understood

File: src/gamelogic/Game.cpp [hunk 31]
```diff
@@ -98,5 +110,49 @@ namespace GameLogic {
...
+    {
+        return this->_time_since_last_frame;
+    }
+
+    void Game::set_new_state(SPTR_IGameState state)
+    {
+        this->_current_state = state;
+    }
+
+    SPTR_IGameState Game::get_credit_state()
```
Comment (waseland) [inline src/gamelogic/Game.cpp]: This was a shitshow but it's done

File: src/gamelogic/Game.hpp [hunk 32]
```diff
@@ -11,26 +11,57 @@
...
     using Engine::WorldPTR;
 
     class Game {
     public:
         Game();
+        ~Game();
         void run();
 
-    private:
-        bool _running;
```
Comment (waseland) [inline src/gamelogic/Game.hpp]: Solved warning, it works ¯\_(ツ)_/¯

File: src/gamelogic/State/CreditState.cpp (hunk 3 above)
Comment (waseland) [inline src/gamelogic/State/CreditState.cpp]: Heeeey, that's pretty good

File: src/gamelogic/State/CreditState.cpp (hunk 1 above)
Comment (waseland) [inline src/gamelogic/State/CreditState.cpp]: Jup ¯\_(ツ)_/¯

File: src/gamelogic/State/CreditState.cpp (hunk 4 above)
Comment (waseland) [inline src/gamelogic/State/CreditState.cpp]: This needs discussing, for now, this works

File: src/gamelogic/State/CreditState.cpp (hunk 5 above)
Comment (waseland) [inline src/gamelogic/State/CreditState.cpp]: Defined in all the states .hpp files

File: src/engine/SDLFacade.cpp (hunk 26 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Fair enough, that makes for a good reason to fix this here.

File: src/engine/SDLFacade.cpp (hunk 25 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp [hunk 33]
```diff
@@ -131,6 +131,56 @@ namespace Engine {
...
+    /// A rect will be drawn from the given coordinate with the given width and height The color is set first, secondly the rect is being drawn.
+    ///
+    /// \param rect_start Coordinate of the start of the rectangle
+    /// \param width Width of the rectangle
+    /// \param height Height of the rectangle
+    /// \param color The color of the rectangle. The color class comes from the engine
+    void SDLFacade::draw_rect(const CoordinateDouble& rect_start, const int width, const int height, const Color& color)
+    {
+        SDL_SetRenderDrawColor(this->_renderer, color.r_value, color.g_value, color.b_value, 255);
+         SDL_Rect r;
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: I fixed this

File: src/engine/SDLFacade.cpp (hunk 27 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp [hunk 34]
```diff
@@ -131,6 +131,56 @@ namespace Engine {
...
+    /// An image will be drawn on the coordinates (Accepts only bmp files)
+    ///
+    /// \param path Path to image (if in res it is: res/imagename.bmp)
+    /// \param coordinates Coordinates of where the image has to be drawn
+    void SDLFacade::draw_image(const std::string path, const CoordinateDouble& coordinates)
+    {
+        const char* imagePath = path.c_str();
+        SDL_Surface* image = SDL_LoadBMP(imagePath);
+        if (image == NULL) //TODO: exception
+        {
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp [hunk 35]
```diff
@@ -131,6 +131,56 @@ namespace Engine {
...
+            cout << "FAILED TO FIND THE IMAGE" << endl;
+            cout << imagePath << endl;
+        } else {
+            SDL_Rect SrcR;
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: I fixed this

File: src/engine/raycasting/Raycasting.hpp (hunk 29 above)
Comment (ThomasvLingen) [inline src/engine/raycasting/Raycasting.hpp]: Nope, but this is an unrelated change regardless of my personal preferences

Comment (ThomasvLingen) [inline src/engine/raycasting/Raycasting.hpp]: Also, fixed

File: src/gamelogic/Game.cpp [hunk 36]
```diff
@@ -67,29 +83,25 @@ namespace GameLogic {
         this->_level->set_player(player);
     }
 
+    Game::~Game() {
```
Comment (ThomasvLingen) [inline src/gamelogic/Game.cpp]: done

File: src/gamelogic/Game.hpp [hunk 37]
```diff
@@ -11,26 +11,57 @@
...
+    using State::SPTR_IGameState;
+    using State::IGameState;
     using Engine::Raycasting;
     using Engine::SDLFacade;
     using Engine::WorldPTR;
 
     class Game {
     public:
         Game();
+        ~Game();
```
Comment (ThomasvLingen) [inline src/gamelogic/Game.hpp]: done

File: src/gamelogic/Game.hpp (hunk 32 above)
Comment (ThomasvLingen) [inline src/gamelogic/Game.hpp]: But this was, and still is, an unrelated change.
I'll let it slide for now though

File: src/gamelogic/Game.hpp [hunk 38]
```diff
@@ -11,26 +11,57 @@
...
         void run();
 
-    private:
-        bool _running;
         SDLFacade _SDL_facade;
```
Comment (ThomasvLingen) [inline src/gamelogic/Game.hpp]: done

File: src/gamelogic/Game.hpp [hunk 39]
```diff
@@ -11,26 +11,57 @@
...
         Raycasting _raycasting_engine;
+
+        int get_time_since_last_frame();
```
Comment (ThomasvLingen) [inline src/gamelogic/Game.hpp]: done

File: src/gamelogic/State/CreditState.cpp [hunk 40]
```diff
@@ -0,0 +1,53 @@
+//
+// Created by sander on 11/17/16.
+//
+
+#include "CreditState.hpp"
+
+using State::CreditState;
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: done

File: src/gamelogic/State/CreditState.cpp [hunk 41]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace State {
+
+    CreditState::CreditState()
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: done

File: src/gamelogic/State/CreditState.cpp [hunk 42]
```diff
@@ -0,0 +1,53 @@
...
+    {
+
+    }
+
+    CreditState::~CreditState() {
```
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: done

File: src/gamelogic/State/CreditState.cpp (hunk 3 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: done

File: src/gamelogic/State/CreditState.cpp (hunk 1 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: That does not solve the problem

File: src/gamelogic/State/CreditState.cpp (hunk 4 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.cpp]: Annotate this somewhere in a Github issue before it's forgotten and never fixed

File: src/gamelogic/State/CreditState.hpp (hunk 2 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/CreditState.hpp]: not fixed

File: src/gamelogic/State/HelpState.cpp (hunk 7 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/HelpState.cpp]: not fixed

File: src/gamelogic/State/IGameState.cpp (hunk 8 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.cpp]: done

File: src/gamelogic/State/IGameState.hpp (hunk 9 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.hpp]: done

File: src/gamelogic/State/IGameState.cpp (hunk 10 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/IGameState.cpp]: done

File: src/gamelogic/State/LoadState.hpp (hunk 11 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.hpp]: not fixed

File: src/gamelogic/State/LoadState.cpp (hunk 12 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.cpp]: not fixed

File: src/gamelogic/State/LoadState.cpp (hunk 13 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.cpp]: not fixed

File: src/gamelogic/State/MenuState.cpp (hunk 20 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: not fixed

File: src/gamelogic/State/MenuState.cpp (hunk 15 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: not fixed * 2

File: src/gamelogic/State/MenuState.cpp (hunk 16 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: not fixed

File: src/gamelogic/State/MenuState.cpp (hunk 17 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: Please respond to these kinds of things so we can look for a solution together

File: src/gamelogic/State/MenuState.cpp (hunk 18 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.cpp]: not fixed

File: src/gamelogic/State/MenuState.hpp (hunk 19 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/MenuState.hpp]: not fixed

File: src/gamelogic/State/PauseState.cpp (hunk 22 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/PauseState.cpp]: not fixed

File: src/gamelogic/State/PauseState.cpp (hunk 23 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/PauseState.cpp]: not fixed

File: src/gamelogic/State/LoadState.cpp (hunk 12 above)
Comment (waseland) [inline src/gamelogic/State/LoadState.cpp]: 2 times as fast loading, this was because it was annoying

File: src/gamelogic/State/MenuState.cpp (hunk 15 above)
Comment (waseland) [inline src/gamelogic/State/MenuState.cpp]: This THis THIs THIS!! AAAH

File: src/gamelogic/State/MenuState.cpp (hunk 16 above)
Comment (waseland) [inline src/gamelogic/State/MenuState.cpp]: It's a kind of magic?

File: src/gamelogic/State/LoadState.cpp (hunk 12 above)
Comment (ThomasvLingen) [inline src/gamelogic/State/LoadState.cpp]: This is a horribly hacky way to do this honestly.

Why not halve the values in your if else ladder instead?
Why even have them as constant values?

It'd be far better to define how long 1 step in the loading bar takes.

**Comment by waseland** [comment]
Please someone :'( review pls pls pls pls

**Comment by jorgdebont** [comment]
Already merged and refactored in #57
